tests/test_visitors/test_ast/test_statements/test_variable_used_outside_of_block.py

```diff
@@ -50,19 +50,23 @@ def function():
 """
 
 
-@pytest.mark.parametrize("example,violation", [
-                         (variable_used_outside_for,
-                             VariableUsedOutsideOfForBlockViolation),
-                         (multiple_variables_defined_by_for,
-                             VariableUsedOutsideOfForBlockViolation),
-                         (variable_used_outside_with,
-                             VariableUsedOutsideOfWithBlockViolation),
-                         (multiple_variables_defined_by_with,
-                             VariableUsedOutsideOfWithBlockViolation)
-                         ]
-                         )
-def test_detect_violations(example, violation, assert_errors, parse_ast_tree, default_options):
+@pytest.mark.parametrize('example,violation', [
+    (variable_used_outside_for, [VariableUsedOutsideOfBlockViolation]),
```

Comment:
Well. There are some positive and some negative cases in there.
```
    (multiple_variables_defined_by_with, [VariableUsedOutsideOfBlockViolation]),
    (valid_for_usage, []),
```

It can be replaced with just count of warnings code supposed to produce (0 or 1 in my current cases), I guess. 
Or I can split it into several test cases

Reply:
It is better to have several test cases for different outputs.